src/main.lua

```diff
@@ -1517,85 +1503,47 @@ addEventingHandler(
 	end
 )
 
-addEventingHandler("totalTokenSupply", function(msg)
-	return msg.Action == "Total-Token-Supply" or msg.Action == ActionMap.TotalSupply -- TODO: remove this once we migrate all downstream apps to the new tag
-end, function(msg)
-	-- add all the balances
-	local totalSupply = 0
-	local circulatingSupply = 0
-	local lockedSupply = 0
-	local stakedSupply = 0
-	local delegatedSupply = 0
-	local withdrawSupply = 0
-	local protocolBalance = balances.getBalance(Protocol)
-	local userBalances = balances.getBalances()
-
-	-- tally circulating supply
-	for _, balance in pairs(userBalances) do
-		circulatingSupply = circulatingSupply + balance
-	end
-	circulatingSupply = circulatingSupply - protocolBalance
-	totalSupply = totalSupply + protocolBalance + circulatingSupply
-
-	-- tally supply stashed in gateways and delegates
-	for _, gateway in pairs(gar.getGatewaysUnsafe()) do
-		totalSupply = totalSupply + gateway.operatorStake + gateway.totalDelegatedStake
-		stakedSupply = stakedSupply + gateway.operatorStake
-		delegatedSupply = delegatedSupply + gateway.totalDelegatedStake
-		for _, delegate in pairs(gateway.delegates) do
-			-- tally delegates' vaults
-			for _, vault in pairs(delegate.vaults) do
-				totalSupply = totalSupply + vault.balance
-				withdrawSupply = withdrawSupply + vault.balance
-			end
-		end
-		-- tally gateway's own vaults
-		for _, vault in pairs(gateway.vaults) do
-			totalSupply = totalSupply + vault.balance
-			withdrawSupply = withdrawSupply + vault.balance
-		end
-	end
-
-	-- user vaults
-	local userVaults = vaults.getVaults()
-	for _, vaultsForAddress in pairs(userVaults) do
-		-- they may have several vaults iterate through them
-		for _, vault in pairs(vaultsForAddress) do
-			totalSupply = totalSupply + vault.balance
-			lockedSupply = lockedSupply + vault.balance
-		end
-	end
-
-	LastKnownCirculatingSupply = circulatingSupply
-	LastKnownLockedSupply = lockedSupply
-	LastKnownStakedSupply = stakedSupply
-	LastKnownDelegatedSupply = delegatedSupply
-	LastKnownWithdrawSupply = withdrawSupply
+-- Reference: https://github.com/permaweb/aos/blob/main/blueprints/token.lua#L264-L280
```

Comment:
I'd be wary of linking to actual line numbers from #main here as they could easily drift over time. could probably link to commit hashes

Reply:
```suggestion
-- Reference: https://github.com/permaweb/aos/blob/eea71b68a4f89ac14bf6797804f97d0d39612258/blueprints/token.lua#L264-L280
```